### PR TITLE
BFD-2566: Move linter to pre push hook

### PR DESCRIPTION
**Pull request by cbrunefearless**
**JIRA Ticket:**
[BFD-2566](https://jira.cms.gov/browse/BFD-2566)

**User Story or Bug Summary:**

User Story:
As a BFD contributor, I want the Java linting step to be moved from the pre-commit to the pre-push hook so that I can avoid the penalty of running the linter on every commit.

AC:
The Java linter executes automatically via pre-push hook
The other functionality of the existing pre-commit hook remains in tact

---

### What Does This PR Do?

Moves the linter to the pre push hook

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] The data dictionary has been updated with any field mapping changes, if any were made.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by lsmitchell — DISMISSED** [review]

**Review by brianburton — APPROVED** [review]
Looking forward to using this!

**Review by mjburling — CHANGES_REQUESTED** [review]
I don't believe this will be quite as simple as lifting the existing shell function from `pre-commit` and writing it into `pre-push` because it operates off of a `git diff` which _may_ or _may not_ include differences when preparing to push. Some of the existing function applies, but it will need to be triggered when there are relevant differences between the local working copy and whatever is stored in the remote.

**Comment by cbrunefearless** [comment]
> I don't believe this will be quite as simple as lifting the existing shell function from `pre-commit` and writing it into `pre-push` because it operates off of a `git diff` which _may_ or _may not_ include differences when preparing to push. Some of the existing function applies, but it will need to be triggered when there are relevant differences between the local working copy and whatever is stored in the remote.

Im doing a compare on the remote and local of the apps directory to see if we should run the formatter, if there is any changes than we should run the formatter if its in the other folders that are not maven directories, do not run the formatter.

**Review by mjburling — APPROVED** [review]

File: apps/pom.xml [hunk 1]
```diff
@@ -586,8 +562,39 @@ checkSecretFilesForPlainText() {
...
+checkJavaFormat() {
+  echo 'Verifying Java source code format...'
+  branch_name="$(git rev-parse --abbrev-ref HEAD 2>&1)"
+
+  git fetch origin 2>&1
+
+  localChanges="$(git diff --name-only "origin/${branch_name}:apps" "${branch_name}:apps" 2>&1)"
+
+  echo "Verifying Java source code format for branch: ${branch_name} ..."
+  if [ "$localChanges" = '' ]; then
```
Comment (mackec1) [inline apps/pom.xml:579]: obviously a matter of preference, but curious...
So if 'no changes detected', why not just exit right there? minimizes 'if' nesting resulting in less indentqation

**Review by mackec1 — APPROVED** [review]

File: apps/pom.xml (hunk 1 above)
Comment (cbrunefearless) [reply apps/pom.xml:579]: we could but didnt want to replicate two return 0's

**Pull request merged**